client/extensions/woocommerce/state/action-list/README.md

```diff
@@ -20,107 +20,115 @@ created.)
+}
+```
 
-It's the responsibility of such a handler to ensure that the appropriate success
-or failure action is dispatched at the conclusion of the asynchronous operation.
+Then step2 is started.
+```
+{
+  prevSteps: [ step1 ],
+  currentStep: step2,
+  nextSteps: [ step3, step4, step5 ],
+}
+```
 
-This should map easily to any sort of promise situation by having the `then` dispatch
-success, and the `catch` dispatch failure.
+And so on...
+```
+{
+  prevSteps: [ step1, step2 ],
+  currentStep: null,
+  nextSteps: [ step3, step4, step5 ],
+}
+```
 
-Assuming the action has a follow-up asynchronous response, it will be handled based on status:
+Until all steps are complete.
+```
+{
+  prevSteps: [ step1, step2, step3, step4, step5 ],
+  currentStep: null,
+  nextSteps: [],
+}
+```
 
-#### Edit State Updates
+### Step objects
 
-It is also the responsibility of the handlers to ensure that the edit state is updated
-appropriately after each step. This means if an API object was created, the state
-should be updated accordingly, for example.
+Each step object consists initially of a description and an `onStep` event.
 
-#### Action List Updates
+```
+{ description: translate( 'Reticulating Splines' ), onStep: reticulateSplines }
+```
 
-For many cases, there will be placeholder ids for things that have not yet been created.
-It's important for each handler to update these placeholder ids with real ones as they
-become available each step along the way.
+Note that the description is designed for user display, so it should be translated.
 
-#### Step End
+The `onStep` function has two parameters that are passed to it:
+```
+function onStep( dispatch, actionList ) {}
+```
 
-The step is given an `endTime` field with a timestamp. This tracks timing for the
-operation as well as status of the step as "complete" if successful, or "failed" if
-an `error` field is present. If there are more steps after this one, the next
-step is started. If this is the last step, then no further action is taken.
+The `dispatch` function can be used to send off actions to the system. These can
+be thunks or asynchronous actions, anything that the store dispatch can handle.
 
-#### Action List Complete
+The `actionList` parameter is the current state of the action list.
 
-The entire list is considered done either when all steps have been ended
-successfully, or when an error is logged on a step. When this occurs, if the
-`successAction` or `failureAction` is designated, the appropriate one will be
-dispatched at the end of action list processing.
+Note that it's the responsibility of `onStep` to eventually dispatch either the
+step success action or step failure action after it is complete. This is crucial
+to keeping the action list moving through its steps. The success or failure action
+can be dispatched either immediately or asynchronously. It doesn't matter when it's
+dispatched, only that it eventually is, or the action list stops.
```

Comment:
I expanded on this, using the names of the action creators and describing what happens with each.